sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONOptions.scala

```diff
@@ -53,6 +55,12 @@ private[sql] class JSONOptions(
   private val parseMode = parameters.getOrElse("mode", "PERMISSIVE")
   val columnNameOfCorruptRecord = parameters.get("columnNameOfCorruptRecord")
 
+  // Share date format object as it is expensive to parse date pattern.
+  val dateFormat: SimpleDateFormat = {
```

Comment:
http://docs.oracle.com/javase/6/docs/api/java/text/SimpleDateFormat.html

Synchronization

Date formats are not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.